Microsoft.DotNet.Interactive/JavaScriptKernel.cs

```diff
@@ -12,8 +12,11 @@ public class JavaScriptKernel : KernelBase
     {
         public const string DefaultKernelName = "javascript";
 
-        public JavaScriptKernel() : base(DefaultKernelName)
+        private string _hostAddress;
+
+        public JavaScriptKernel(string hostAddress) : base(DefaultKernelName)
```

Comment:
I think the address should be moved to the `FrontEndEnvironment`. It shouldn't be the responsibility of someone creating the kernel to understand details of the environment like this, and the wrapper script(s) will likely need to vary in other ways by frontend anyway.

Reply:
What are your thoughts on threading the value through?  `JavaScriptKernel.cs` is in the `Microsoft.DotNet.Interactive` project, but `JupyterFrontendEnvironment` (which seems like the logical place for this) is one level up in `Microsoft.DotNet.Interactive.Jupyter`.  The other option is to add `HttpEndpoint` to `FrontendEnvironmentBase`, but that seems heavy-handed to force all front ends to support the existence of an http endpoint.

Reply:
The point of the `FrontEndEnvironment` is to vary by capabilities, so it could be added to the `JupyterFrontEndEnvironment` but not to `FrontendEnvironmentBase`.

I don't think the `Microsoft.DotNet.Interactive` layer should be aware of the HTTP APIs.

Reply:
The port is the port of dotnet interactive, it is not related to the frontend. Not so sure the value should be there. If it does then is part of the base class not just jupyter

Reply:
The purpose of the `FrontEndEnvironment` (though it has no significant behavior currently) was to allow the kernel to detect front-end capabilities and vary output accordingly, e.g. script loading capabilities. This is very much in line with that goal.

Reply:
My point above is that we can't add it to `JupyterFrontEndEnvironment` because of this:

``` C#
// JavaScriptKernel.cs in top-level project `Microsoft.DotNet.Interactive`
if (context.FrontendEnvironment is JupyterFrontendEnvironment j) // <- this doesn't work because `JupyterFrontendEnvironment is defined (appropriately) in `Microsoft.DotNet.Interactive.Jupyter`
{
    // ...
}
```

We either have to:

1. Move `JavaScriptKernel.cs` to `Microsoft.DotNet.Interactive.Jupyter` so that having a reference to `JupyterFrontendEnvironment` makes sense, or
2. Add the host (e.g., `http://localhost:1234`) to `FrontenvEnvironmentBase`, or
3. Add `class FooFrontEndEnvironment : FrontendEnvironmentBase { string Host {get; set;} }` to `Microsoft.DotNet.Interactive`.

Option (1) possibly makes sense because we don't actually have a JavaScript kernel; it's dependent on the host understanding `<script>` tags.  The case where this doesn't make sense is when we have a browser-based front end that's not Jupyter (e.g., Azure notebooks).  Options (2) or (3) I think I like best because it's entirely conceivable for a command-line interactive environment to have http language services.

I'm leaning towards option 3.